### ISSUE TITLE
Knock Knock !!!

### ISSUE THREAD

**Issue by aiamuzz** [Sylvain303/move-to-next-monitor#1]
@Sylvain303 ... thanks for the modification of this tool from @jc00ke's previous work ...

While @jc99ke's effort did move the windows it was scattering the moved windows when they were not maximized, Maximized windows were moving maximized as intended ...

thanks to you efforts now i can move even the unmaximized windows with acceptable results ...

Are you active ?

I had a few questions about your script ...

thanks.

**Comment by Sylvain303** [comment]
hello @aiamuzz 

Well, I'm not very active you're right. :unamused: 

As you may have seen, not so well updated script here.

I need something similar (moving windows) around on my new configuration, 38" above laptop.
So I may reboot the script. I plan to rewrite it in python with sub-process call as from bash, but without the hassle of managing array in bash.

On what OS are you  running?

**Comment by aiamuzz** [comment]
I am running ... LINUX DEEPIN ...

**Comment by aiamuzz** [comment]
> As you may have seen, not so well updated script here.
not true, mainly becoz it is moving resized windows ... its a workable solution !!!

> 38" above laptop.
38" LAPTOP ? ... i suppose you meant to say LED monitor ... :)

> So I may reboot the script. I plan to rewrite it in python with sub-process call as from bash, but without the hassle of managing array in bash.

some observations ... if they are within the scope of the packages you rely upon ...

1. Keyboard Shortcut key works like a charm.
2. Maximized windows move between my external monitor(22") without any problem
3. FULL SCREEN APPLICATION(MOSTLY VIDEO) DO NOT MOVE, would be nice if they too could move full screen to full screen across monitors.
4. An application window launched on the laptop screen moves to the larger monitor without much of an aberration, whereas an application window launched on the larger monitor moves to the laptop(smaller) screen with its placement & dimensions falling outside(awkwardly) the laptop screens dimensions ... a not so elegant switch.
- IF ONLY THE RELIED UPON PACKAGES AND SCRIPT COULD SOMEHOW CALCULATE THE RELATIVE POSITION(PLACEMENT) OF ANY WINDOW ON A GIVEN SCREEN AND MOVE IT PROPORTIONALLY TO THE WIDTH AND HEIGHT OF THE SCREENS ... THEN ... THAT WOULD CERTAINLY BE A VERY ELEGANT IMPLEMENTATION !!!

 If the current windows screen geometry relative to the screen its present on, could be ascertained somehow then the the window could be sent to a placement that is proportional to the screen geometry of the larger screen.
 
not to scare you but just putting it out there ... :)

**Comment by Sylvain303** [comment]
> LINUX DEEPIN

Is it based on xfce? or some other window manager?

Don't you think such tool: `move-to-next-monitor` script doesn't have been super-seed by some native windowmanager feature now?

I suppose it doesn't has you've followed all the link until this repository. :wink: 

I also have code that could fit in the tool:

* move resized window (half or 1/4 of the screen) to the  left/right/top/bottom

Do you think it should stay in bash script (require high bash scripting level quite rare) or should it move to some more "programming" language with float computation, etc? :wink:

**Comment by Sylvain303** [comment]
> > 38" above laptop.
> 38" LAPTOP ? ... i suppose you meant to say LED monitor ... :)

hum, something like that:

![image](https://user-images.githubusercontent.com/274026/210764776-57286837-fa76-47ca-862f-c242f42b205f.png)

**Comment by aiamuzz** [comment]
> > > 38" above laptop.
> > > 38" LAPTOP ? ... i suppose you meant to say LED monitor ... :)
> 
> hum, something like that:
> 
> ![image](https://user-images.githubusercontent.com/274026/210764776-57286837-fa76-47ca-862f-c242f42b205f.png)

wow ! ... awesome monitor size ... a 38" diagonal must give you some much of real estate ...

**Comment by Sylvain303** [comment]
> 38" monitor

Yeah that's big, but there exist bigger screen now.

Obviously it requires also a large desk, too. 38" is a bit like 2x 27" glued together without black interruption in the middle of the screen. So it allow middle screen window, in front of you. Full screen is a bit too large 3840x1600 pixel. 2 half width window I quite useful. One half size in the center is good too, this left room for 2x 1/4 width sized window on each side:

```asciiart
+------------+-------------------------+------------+                                                                                                
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
|            |                         |            |
+------------+-------------------------+------------+
```
 

Do you have programming skill? Do you want to participate on the enhancement you suggested?
